iot-hub/Samples/device/DeviceReconnectionSample/DeviceReconnectionSample.cs

```diff
@@ -56,30 +57,32 @@ public DeviceReconnectionSample(List<string> deviceConnectionStrings, TransportT
 
         public async Task RunSampleAsync(TimeSpan sampleRunningTime)
         {
-            using var cts = new CancellationTokenSource(sampleRunningTime);
+            s_cancellationTokenSource = new CancellationTokenSource(sampleRunningTime);
             Console.CancelKeyPress += (sender, eventArgs) =>
             {
                 eventArgs.Cancel = true;
-                cts.Cancel();
+                s_cancellationTokenSource.Cancel();
                 _logger.LogInformation("Sample execution cancellation requested; will exit.");
             };
 
             _logger.LogInformation($"Sample execution started, press Control+C to quit the sample.");
 
             try
             {
-                await InitializeAndOpenClientAsync();
-                await Task.WhenAll(SendMessagesAsync(cts.Token), ReceiveMessagesAsync(cts.Token));
+                await InitializeAndSetupClientAsync(s_cancellationTokenSource.Token);
+                await Task.WhenAll(SendMessagesAsync(s_cancellationTokenSource.Token), ReceiveMessagesAsync(s_cancellationTokenSource.Token));
             }
             catch (Exception ex)
             {
                 _logger.LogError($"Unrecoverable exception caught, user action is required, so exiting...: \n{ex}");
+                s_cancellationTokenSource.Cancel();
```

Comment:
Probably need to wait for the above tasks to complete before disposing and moving on, no?